yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -1289,3 +1289,115 @@ func (yb *TargetYugabyteDB) GetImportedSnapshotRowCountForTable(tableName string
+		defer rows.Close()
+		for rows.Next() {
+			var colName string
+			err = rows.Scan(&colName)
+			if err != nil {
+				log.Errorf("error in scanning identity column name: %v", err)
+				return false, fmt.Errorf("error in scanning identity column name: %w", err)
+			}
+			identityColumns = append(identityColumns, colName)
+		}
+		return false, nil
+	})
+	return identityColumns, err
+}
+
+func (yb *TargetYugabyteDB) GetGeneratedAlwaysAsIdentityColumnsForTables(tables []string) (map[string][]string, error) {
+	var result = make(map[string][]string)
+	for _, table := range tables {
+		identityColumns, err := yb.GetGeneratedAlwaysIdentityColumnNamesForTable(table)
+		if err != nil {
+			log.Errorf("error in getting identity columns for table %s: %v", table, err)
+			return nil, fmt.Errorf("error in getting identity columns for table %s: %w", table, err)
+		}
+		if len(identityColumns) > 0 {
+			log.Infof("identity columns for table %s: %v", table, identityColumns)
+			result[table] = identityColumns
+		}
+	}
+	return result, nil
+}
+
+func (yb *TargetYugabyteDB) DisableGeneratedAlwaysAsIdentityColumns(tableColumnsMap map[string][]string) error {
+	log.Infof("disabling generated always as identity columns")
+	for table, columns := range tableColumnsMap {
+		qualifiedTblName := yb.qualifyTableName(table)
+		batch := pgx.Batch{}
+		for _, column := range columns {
+			query := fmt.Sprintf(`ALTER TABLE %s ALTER COLUMN %s SET GENERATED BY DEFAULT`, qualifiedTblName, column)
+			batch.Queue(query)
+		}
+		err := yb.connPool.WithConn(func(conn *pgx.Conn) (bool, error) {
+			br := conn.SendBatch(context.Background(), &batch)
+			for i := 0; i < batch.Len(); i++ {
+				_, err := br.Exec()
+				if err != nil {
+					log.Errorf("error executing query to alter always identity columns for table(%s): %v", qualifiedTblName, err)
+					return false, fmt.Errorf("error executing query to alter always identity columns for table(%s): %w", qualifiedTblName, err)
+				}
+			}
+			if err := br.Close(); err != nil {
+				log.Errorf("error closing batch of queries to alter alwayss identity columns for table(%s): %v", qualifiedTblName, err)
+				return false, fmt.Errorf("error closing batch of queries to alter always identity columns for table(%s): %w", qualifiedTblName, err)
+			}
+			return false, nil
+		})
+		if err != nil {
+			log.Errorf("error in disabling identity columns for table %s: %v", qualifiedTblName, err)
+			return fmt.Errorf("error in disabling identity columns for table %s: %w", qualifiedTblName, err)
+		}
+	}
+	return nil
+}
+
+func (yb *TargetYugabyteDB) EnableGeneratedAlwaysAsIdentityColumns(tableColumnsMap map[string][]string) error {
+	log.Infof("enabling generated always as identity columns")
+	for table, columns := range tableColumnsMap {
+		qualifiedTblName := yb.qualifyTableName(table)
+		batch := pgx.Batch{}
+		for _, column := range columns {
+			query := fmt.Sprintf(`ALTER TABLE %s ALTER COLUMN %s SET GENERATED ALWAYS`, qualifiedTblName, column)
```

Comment:
Except for the `SET GENERATED ALWAYS` (and log messages) this function is the same as `Disable*` counterpart.

How about having a private method `alterColumns()` function that will be called as:
```
// From Disable*()
err = yb.alterColumns(tableColumnsMap, "SET GENERATED BY DEFAULT")

// From Enable*()
err = yb.alterColumns(tableColumnsMap, "SET GENERATED ALWAYS")
```

Reply:
nice idea, done